components/sts/org.wso2.carbon.identity.sts.passive.ui/src/main/java/org/wso2/carbon/identity/sts/passive/ui/PassiveSTS.java

```diff
@@ -441,6 +444,10 @@ private void sendFrameworkForLogout(HttpServletRequest request, HttpServletRespo
         authenticationRequest.setRequestQueryParams(request.getParameterMap());
         authenticationRequest.setCommonAuthCallerPath(selfPath);
         authenticationRequest.appendRequestQueryParams(request.getParameterMap());
+        // According to ws-federation-1.2-spec; 'wtrealm' will not be sent in the Passive STS Logout Request.
+        if (sessionDTO.getRealm() == null || sessionDTO.getRealm().trim().length() == 0) {
+            authenticationRequest.setRelyingParty(" ");
+        }
```

Comment:
In the Passive STS logout flow, authentication framework expects a "relyingParty" to be set in the passive request (even though it's not specified in the WT-Trust Spec), which will be retrieved by the  DefaultRequestCoordinator class in the framework. When analyzing the Passive STS login flow , "relyingParty" value was equaled to the "relam" value sent by the Passive STS request. Correspondingly, in the STS logout flow "replying party" value is set to an empty string in the request which will be sent to authentication framework. (i.e Refer https://github.com/kasungayan/carbon-identity/blob/PassiveSTS/components/sts/org.wso2.carbon.identity.sts.passive.ui/src/main/java/org/wso2/carbon/identity/sts/passive/ui/PassiveSTS.java#L298 which sets the relying party in the passive sts login request as the relam value retrieved from session DTO object )